src/server.js

```diff
@@ -267,19 +267,20 @@ function localAuth(client, user, password, callback) {
 		return callback(false);
 	}
 
-	var result = Helper.password.compare(password, client.config.password);
+	var result = Helper.password.compareAsync(password, client.config.password);
 
-	if (result && Helper.password.requiresUpdate(client.config.password)) {
-		var hash = Helper.password.hash(password);
+	result.then((matching) => {
+		if (Helper.password.requiresUpdate(client.config.password)) {
+			var hash = Helper.password.hash(password);
 
-		client.setPassword(hash, function(success) {
-			if (success) {
-				log.info(`User ${colors.bold(client.name)} logged in and their hashed password has been updated to match new security requirements`);
-			}
-		});
-	}
-
-	return callback(result);
+			client.setPassword(hash, function(success) {
+				if (success) {
+					log.info(`User ${colors.bold(client.name)} logged in and their hashed password has been updated to match new security requirements`);
+				}
+			});
+		}
+		callback(matching);
+	}, (error) => log.error(`Error while checking users password. Error: ${error}`));
```

Comment:
No need for parentheses around `error`, unless we set up ESLint to be strict like that.